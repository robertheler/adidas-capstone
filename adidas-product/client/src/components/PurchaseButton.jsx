import React, { Component, Fragment} from 'react';
import styled from 'styled-components';

const RenderDiv = styled.div`
  font-size: 13px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  text-transform: uppercase;
  display: flex;
  justify-content: space-between;
  margin: 35px 0;
`;

const StyledButton = styled.button`
  font-size: 13px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  padding: 15px;
  text-transform: uppercase;
  height: 50px;
  width: 80%;
  background-color: black;
  color: white;
  vertical-align: middle;
  font-weight: 700;
  letter-spacing: 2px;
  display: flex;
  justify-content: space-between;
  border: none;
  position: relative;
  top: -2px;
  box-shadow: 2px 2px 0 0px white, 2px 2px 0 1px black;
  box-sizing: border-box;
  &:active {
    box-shadow: none;
  }

  cursor: pointer;
  &:hover {
    color: gray;
  }
  &:active {
    outline: none;
    transform: translateY(2px) translateX(2px);
    box-shadow: none;
  };
`
StyledButton.displayName = 'StyledButton'; // to be used in testing

const Favorite = styled.div`
  width: 50px;
  height: 50px;
  border: 1px solid black;
  box-sizing: border-box;
  cursor: pointer;
  &:hover {
    border: 1px solid gray;
    color: gray;
  };
  &:active {
    transform: translateY(2px) translateX(2px);
  }
  vertical-align: middle;
`

const StyledSvg = styled.svg`
  margin: 0px;
  height: 20px;
  width: 20px;
  padding: 14px;
  &:hover {
    color: gray;
  };
`

const StyledPath = styled.path`
  fill:${(props) => props.favorite === 'true'? '#e32b2b' : 'none'}};
  stroke: ${(props) => props.favorite === 'true'? '#e32b2b' : 'black'}};
  stroke-width: 2;
  strokeMiterlimit: 10;
  d: path("M7.38 6H4.42L2 10l8 8 8-8-2.41-4h-2.98L10 9 7.38 6z");
  pointer-events: all;
`

const StyledPathArrow = styled.path`
  padding: 0px;
  margin-top: 0px;
  stroke-linecap: square;
  stroke-width: 2px;
  d: path(${props => props.d});
`
StyledPath.displayName = 'StyledPath';
class PurchaseButton extends Component {
  constructor(props){
    super(props);
    this.state = {
      favorite: false
    }

    this.favorite =  this.favorite.bind(this);
  }

  purchase(event){
    event.currentTarget.blur();
  }

  favorite(event){
    this.setState({
      favorite: ! this.state.favorite
    })
  }

  render() {
    return (
        <RenderDiv>
          <StyledButton onClick={this.purchase}>
            <span>Add to bag</span>

              <svg height="20px" width="25px">
                <g fill="none" stroke="currentColor" strokeMiterlimit="10">
                  <StyledPathArrow d="M17.59 7l5 5-5 5M0 12h22"></StyledPathArrow>
                </g>
              </svg>

          </StyledButton>
          <Favorite onClick={this.favorite}>
            <StyledSvg>
            <StyledPath favorite={this.state.favorite.toString()}></StyledPath>
            </StyledSvg>
          </Favorite>
        </RenderDiv>

    );
  }
}

export default PurchaseButton;