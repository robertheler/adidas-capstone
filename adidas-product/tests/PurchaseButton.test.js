import React from 'react';
import { shallow, mount, render } from 'enzyme';
import PurchaseButton from '../client/src/components/PurchaseButton';
import StyledButton from '../client/src/components/PurchaseButton';
import StyledPath from '../client/src/components/PurchaseButton';
import 'jest-styled-components';
// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0]
  };
  const wrapper = shallow(<PurchaseButton color={props.color} product={props.product} isSelected={props.isSelected} handleColorChange={props.handleColorChange}/>);

  const mockEventGenerator = (str) => {
    return {
      currentTarget: {
        name: 'todo',
        value: str,
        blur: () => {}
      },
      preventDefault: () => {},
    };
  };
  return { wrapper, props, mockEventGenerator };

}

describe('Title Component Test Suite', () => {
  const { wrapper, props, mockEventGenerator } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('It should toggle favorited state on click' ,() => {
    wrapper.instance().state.favorite = false;
    expect(wrapper.instance().state.favorite).toBe(false);
    wrapper.instance().favorite();
    expect(wrapper.instance().state.favorite).toBe(true);
  });

  // it('It should trigger purchase when clicked', () => {
  //   wrapper.instance().purchase = jest.fn();
  //   wrapper.find('StyledButton').simulate('click', mockEventGenerator);
  //   expect(wrapper.instance().purchase).toHaveBeenCalled();
  // })

  it('It should render a filled heart if favorited', () => {
    let heart = mount(<StyledPath favorite="true"/>)
    expect(heart).toMatchSnapshot();
  });

  it('It should render an empty heart if not favorited', () => {
    let heart = shallow(<StyledPath favorite="false"/>)
    expect(heart).toMatchSnapshot();
  });

});