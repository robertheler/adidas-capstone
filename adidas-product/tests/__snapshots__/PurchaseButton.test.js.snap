// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Test Suite It should render a filled heart if favorited 1`] = `
.c0 {
  font-size: 13px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 35px 0;
}

.c1 {
  font-size: 13px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  padding: 15px;
  text-transform: uppercase;
  height: 50px;
  width: 80%;
  background-color: black;
  color: white;
  vertical-align: middle;
  font-weight: 700;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: none;
  position: relative;
  top: -2px;
  box-shadow: 2px 2px 0 0px white,2px 2px 0 1px black;
  box-sizing: border-box;
  cursor: pointer;
}

.c1:active {
  box-shadow: none;
}

.c1:hover {
  color: gray;
}

.c1:active {
  outline: none;
  -webkit-transform: translateY(2px) translateX(2px);
  -ms-transform: translateY(2px) translateX(2px);
  transform: translateY(2px) translateX(2px);
  box-shadow: none;
}

.c3 {
  width: 50px;
  height: 50px;
  border: 1px solid black;
  box-sizing: border-box;
  cursor: pointer;
  vertical-align: middle;
}

.c3:hover {
  border: 1px solid gray;
  color: gray;
}

.c3:active {
  -webkit-transform: translateY(2px) translateX(2px);
  -ms-transform: translateY(2px) translateX(2px);
  transform: translateY(2px) translateX(2px);
}

.c4 {
  margin: 0px;
  height: 20px;
  width: 20px;
  padding: 14px;
}

.c4:hover {
  color: gray;
}

.c5 {
  fill: none;
  stroke: black;
  stroke-width: 2;
  strokeMiterlimit: 10;
  d: path("M7.38 6H4.42L2 10l8 8 8-8-2.41-4h-2.98L10 9 7.38 6z");
  pointer-events: all;
}

.c2 {
  padding: 0px;
  margin-top: 0px;
  stroke-linecap: square;
  stroke-width: 2px;
  d: path(M17.59 7l5 5-5 5M0 12h22);
}

<PurchaseButton
  favorite="true"
>
  <styled.div>
    <div
      className="c0"
    >
      <StyledButton
        onClick={[Function]}
      >
        <button
          className="c1"
          onClick={[Function]}
        >
          <span>
            Add to bag
          </span>
          <svg
            height="20px"
            width="25px"
          >
            <g
              fill="none"
              stroke="currentColor"
              strokeMiterlimit="10"
            >
              <styled.path
                d="M17.59 7l5 5-5 5M0 12h22"
              >
                <path
                  className="c2"
                  d="M17.59 7l5 5-5 5M0 12h22"
                />
              </styled.path>
            </g>
          </svg>
        </button>
      </StyledButton>
      <styled.div
        onClick={[Function]}
      >
        <div
          className="c3"
          onClick={[Function]}
        >
          <styled.svg>
            <svg
              className="c4"
            >
              <StyledPath
                favorite="false"
              >
                <path
                  className="c5"
                />
              </StyledPath>
            </svg>
          </styled.svg>
        </div>
      </styled.div>
    </div>
  </styled.div>
</PurchaseButton>
`;

exports[`Title Component Test Suite It should render an empty heart if not favorited 1`] = `
<styled.div>
  <StyledButton
    onClick={[Function]}
  >
    <span>
      Add to bag
    </span>
    <svg
      height="20px"
      width="25px"
    >
      <g
        fill="none"
        stroke="currentColor"
        strokeMiterlimit="10"
      >
        <styled.path
          d="M17.59 7l5 5-5 5M0 12h22"
        />
      </g>
    </svg>
  </StyledButton>
  <styled.div
    onClick={[Function]}
  >
    <styled.svg>
      <StyledPath
        favorite="false"
      />
    </styled.svg>
  </styled.div>
</styled.div>
`;

exports[`Title Component Test Suite It should render correctly 1`] = `
<styled.div>
  <StyledButton
    onClick={[Function]}
  >
    <span>
      Add to bag
    </span>
    <svg
      height="20px"
      width="25px"
    >
      <g
        fill="none"
        stroke="currentColor"
        strokeMiterlimit="10"
      >
        <styled.path
          d="M17.59 7l5 5-5 5M0 12h22"
        />
      </g>
    </svg>
  </StyledButton>
  <styled.div
    onClick={[Function]}
  >
    <styled.svg>
      <StyledPath
        favorite="false"
      />
    </styled.svg>
  </styled.div>
</styled.div>
`;
